### ISSUE TITLE
error  Unknown property 'mask-type' found  react/no-unknown-property

### ISSUE THREAD

**Issue by JustFly1984** [jsx-eslint/eslint-plugin-react#3466]
The property in question is on svg element, so may be add it to allowed list?

**Comment by ljharb** [comment]
Can you share the jsx for this? I see `mask` on MDN, and `mask-type` CSS, but not a `mask-type` attribute.

I do see https://github.com/gregberge/svgr/issues/336 which suggests it *might* apply to `<mask>` elements though?

**Comment by Simek** [comment]
Just encounter same issue today after upgrading the lint stack in one of the projects (the property seems to work correctly).

I think that the problem has its root in `@types/react` package, where this `maskType` property is not defined as a valid attribute for SVG elements:
* https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L2648-L2650

**Comment by sandgraham** [comment]
Just chiming in- saw this error pop up on some inline SVG like so:
```xml
<mask
  mask-type="alpha"
  maskUnits="userSpaceOnUse"
  x="100"
  y="150"
  width="100"
  height="150"
>
  <path d="..." />
</mask>
```
Note, changing the property to `maskType` also reveals that it isn't a type on lib.dom `SVGMaskElement` yet.

The draft spec [defines](https://drafts.fxtf.org/css-masking/#the-mask-type) it as applying to mask elements and, notably, as a "presentation attribute", which, from what I can tell in the [SVG 1.1 spec](https://www.w3.org/TR/2011/REC-SVG11-20110816/styling.html#UsingPresentationAttributes), should be assignable as an inline attribute.

> ...there is a corresponding XML attribute (the presentation attribute) with the same name that is available on all relevant SVG elements

**Comment by ljharb** [comment]
That "corresponding" stuff is very, very confusing. It'd be great to get confirmation from someone who works on the SVG specs.

**Comment by sandgraham** [comment]
I filed an issue with W3C to try and get some clarification: https://github.com/w3c/fxtf-drafts/issues/480. Let me know if I missed something!
